vncviewer/Viewport.cxx

```diff
@@ -122,7 +122,9 @@ Viewport::Viewport(int w, int h, const rfb::PixelFormat& /*serverPF*/, CConn* cc
     altGrArmed(false),
 #endif
     firstLEDState(true), pendingClientClipboard(false),
-    menuCtrlKey(false), menuAltKey(false), cursor(nullptr)
+    menuCtrlKey(false), menuAltKey(false), cursor(nullptr),
+    useSystemCursorIfBlank(alwaysCursor && !strcasecmp("system", cursorType)),
```

Comment:
The fallback logic here is different from the options dialog. Here you get the dot cursor for invalid values, whilst in the options dialog you get the system cursor for invalid values.

For sanity, it might be best if the fallback is the same as the default value.